### ISSUE TITLE
Tensorflow calculates incorrect loss for `tf.keras` models when using Weights.

### ISSUE THREAD

**Issue by skiler07** [tensorflow/tensorflow#23767]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (No):
- OS Platform and Distribution (e.g., MacOS + Google Cloud VMs):
- TensorFlow installed from (source):
- TensorFlow version (1.10 and 1.12):
- Python version: 3.5

**Describe the current behavior**
The loss calculation is not correct when working with `tf.keras.`. After building the model, [`tf.keras.fit_generator`](https://www.tensorflow.org/api_docs/python/tf/keras/models/Model) should accept `(inputs, targets, sample_weights)` as inputs. However, if I multiply the `sample_weights` by 10000,  the loss doesn't change.

The bug seems to appear from 1.10 version of Tensorflow onwards e.g. (1.11, 1.12)

**Describe the expected behavior**
If I increase the weighting by a certain factor, the overall loss of the model should increase by the same factor. Given the model is doing random guessing. 

**Code to reproduce the issue**
```
import numpy as np
import tensorflow as tf

WEIGHT_VARIABLE = 1

no_of_features = 10
timesteps = 3
batch_size = 32

def data_gen():

    while True:
        numerical = np.random.randint(5, size=(batch_size, timesteps, no_of_features))
        y = np.random.randint(2, size=batch_size)
        w = np.ones(batch_size) * WEIGHT_VARIABLE

        yield {'numeric_input': numerical}, y, w


def build_model():
    numerical_input = tf.keras.layers.Input(shape=(timesteps, no_of_features), name='numeric_input')
    rnn_out = tf.keras.layers.GRU(32, return_sequences=False)(numerical_input)
    dense = tf.keras.layers.Dense(1, activation='sigmoid', name='main_output')(rnn_out)

    model = tf.keras.models.Model(numerical_input, dense)

    params = {
        'loss': 'binary_crossentropy',
        'optimizer': tf.keras.optimizers.Adam(),
        'metrics': [tf.keras.metrics.binary_crossentropy, tf.keras.metrics.binary_accuracy]
    }
    model.compile(**params)

    return model


def train_model():
    gen1 = data_gen()
    model = build_model()

    model.fit_generator(gen1, epochs=30, steps_per_epoch=10)


if __name__ == '__main__':
    train_model()
```

In the above code, you simply need to change the `WEIGHT_VARIABLE = 1` From 1 to 100000 and rerun the file.

**Other info / logs**
### v1.10

`WEIGHT_VARIABLE = 1`
10/10 [==============================] - 1s 128ms/step - loss: 0.7407 - binary_crossentropy: 0.7407 - binary_accuracy: 0.5031
Epoch 2/5
10/10 [==============================] - 0s 4ms/step - loss: 0.7043 - binary_crossentropy: 0.7043 - binary_accuracy: 0.5125
Epoch 3/5
10/10 [==============================] - 0s 4ms/step - loss: 0.7055 - binary_crossentropy: 0.7055 - binary_accuracy: 0.5219
Epoch 4/5
10/10 [==============================] - 0s 4ms/step - loss: 0.7002 - binary_crossentropy: 0.7002 - binary_accuracy: 0.5250
Epoch 5/5
10/10 [==============================] - 0s 4ms/step - loss: 0.6944 - binary_crossentropy: 0.6944 - binary_accuracy: 0.5375

`WEIGHT_VARIABLE = 10000`

10/10 [==============================] - 1s 131ms/step - loss: 7235.5976 - binary_crossentropy: 0.7236 - binary_accuracy: 0.4562
Epoch 2/5
10/10 [==============================] - 0s 4ms/step - loss: 7271.9184 - binary_crossentropy: 0.7272 - binary_accuracy: 0.4844
Epoch 3/5
10/10 [==============================] - 0s 4ms/step - loss: 7276.9147 - binary_crossentropy: 0.7277 - binary_accuracy: 0.4500
Epoch 4/5
10/10 [==============================] - 0s 4ms/step - loss: 7052.0121 - binary_crossentropy: 0.7052 - binary_accuracy: 0.4625
Epoch 5/5
10/10 [==============================] - 0s 4ms/step - loss: 7187.0285 - binary_crossentropy: 0.7187 - binary_accuracy: 0.4969

### v1.12

`WEIGHT_VARIABLE = 1`
10/10 [==============================] - 1s 68ms/step - loss: 0.7188 - binary_crossentropy: 0.7188 - binary_accuracy: 0.5312
Epoch 2/5
10/10 [==============================] - 0s 4ms/step - loss: 0.7044 - binary_crossentropy: 0.7044 - binary_accuracy: 0.4969
Epoch 3/5
10/10 [==============================] - 0s 4ms/step - loss: 0.7086 - binary_crossentropy: 0.7086 - binary_accuracy: 0.4844
Epoch 4/5
10/10 [==============================] - 0s 4ms/step - loss: 0.7075 - binary_crossentropy: 0.7075 - binary_accuracy: 0.4500
Epoch 5/5
10/10 [==============================] - 0s 4ms/step - loss: 0.6950 - binary_crossentropy: 0.6950 - binary_accuracy: 0.5187

`WEIGHT_VARIABLE = 10000`

10/10 [==============================] - 1s 74ms/step - loss: 0.9084 - binary_crossentropy: 0.9084 - binary_accuracy: 0.4719
Epoch 2/5
10/10 [==============================] - 0s 4ms/step - loss: 0.7120 - binary_crossentropy: 0.7120 - binary_accuracy: 0.5062
Epoch 3/5
10/10 [==============================] - 0s 4ms/step - loss: 0.7024 - binary_crossentropy: 0.7024 - binary_accuracy: 0.5344
Epoch 4/5
10/10 [==============================] - 0s 4ms/step - loss: 0.7257 - binary_crossentropy: 0.7257 - binary_accuracy: 0.4500
Epoch 5/5
10/10 [==============================] - 0s 4ms/step - loss: 0.7013 - binary_crossentropy: 0.7013 - binary_accuracy: 0.4844

**Comment by brge17** [comment]
I can also confirm this issue, looks like it was introduced in ``Tensorflow==1.11.0``

The same issue effects class weights too:

```python
import numpy as np
from tensorflow.python.keras.layers import Input, Dense
from tensorflow.python.keras.models import Model
batch_size = 10
num_input = 5
num_output = 2
positive_weight = 100
def _to_one_hot(y, num_classes):
    """ Compute one-hot encoding of an array. """
    y_one_hot = np.zeros((len(y), num_classes)).astype(np.float32)
    y_one_hot[np.arange(len(y)), y.squeeze()] = 1
    return y_one_hot
class DummyGenerator(object):
    def class_weights(self):
        while True:
            X = np.random.rand(batch_size, num_input)
            y = np.random.randint(num_output, size=batch_size)
            y = _to_one_hot(y, 2)
            yield X, y
    def sample_weights(self):
        while True:
            X = np.random.rand(batch_size, num_input)
            y = np.random.randint(num_output, size=batch_size)
            w = y * (positive_weight - 1)
            y = _to_one_hot(y, 2)
            yield X, y, w
def dummy_model():
    inputs = Input(shape=(num_input,))
    outputs = Dense(2, activation="softmax")(inputs)
    model = Model(inputs=inputs, outputs=outputs)
    model.compile(loss="categorical_crossentropy", optimizer="adam")
    return model
model = dummy_model()
gen = DummyGenerator()
print("Training with sample_weights")
model.fit_generator(gen.sample_weights(), steps_per_epoch=1, epochs=1)
print("Training with class_weights")
model.fit_generator(gen.class_weights(), steps_per_epoch=1, epochs=1, class_weight={0:1, 1:positive_weight})
```

**Comment by pdbaines** [comment]
Confirmed that we ran into the same issue. 👍

**Comment by jgv7** [comment]
Same issue here 👍

**Comment by brge17** [comment]
Can we get an acknowledgement of this issue? I know @fchollet is very busy, maybe @pavithrasv could give it a quick look?

**Comment by shijianjian** [comment]
I do not think it has been fixed. I can still see it happens in 1.14 and even tf2.0 beta. I think it was fixed in 1.13 but re-introduced afterwards.
